ScenarioSetup.ttslua

```diff
@@ -16,6 +16,42 @@ function onLoad(save_state)
     initializeSetupButtons()
 end
 
+function onDrop(player_color)
+    -- Follow the dropped map until it lands.
+    Wait.frames(followLanding, 1)

```

Comment:
Rather than Wait.frames(), I would suggest Wait.condition()

Wait.condition('setGridParameters', function() return not self.resting end)


Also define

function setGridParameters()
    if gridData ...
end